relay-dynamic-config/src/metrics.rs

```diff
@@ -124,3 +125,178 @@ impl TransactionMetricsConfig {
         self.version > 0 && self.version <= TRANSACTION_EXTRACT_VERSION
     }
 }
+
+/// Configuration for generic extraction of metrics from all data categories.
+#[derive(Clone, Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MetricExtractionConfig {
+    /// Versioning of metrics extraction. Relay skips extraction if the version is not supported.
+    pub version: u16,
+
+    /// A list of metric specifications to extract.
+    #[serde(default)]
+    pub metrics: Vec<MetricSpec>,
+
+    /// A list of tags to add to previously extracted metrics.
+    ///
+    /// These tags add further tags to a range of metrics. If some metrics already have a matching
+    /// tag extracted, the existing tag is left unchanged.
+    #[serde(default)]
+    pub tags: Vec<TagMapping>,
+}
+
+impl MetricExtractionConfig {
+    /// Returns `true` if metric extraction is disabled.
+    pub fn is_empty(&self) -> bool {
+        self.version == 0 || (self.metrics.is_empty() && self.tags.is_empty())
+    }
+}
+
+/// Specification for a metric to extract from some data.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MetricSpec {
+    /// Category of data to extract this metric for.
+    pub category: DataCategory,
+
+    /// Identifier of the metric to extract.
```

Comment:
nit: same as in [here](https://github.com/getsentry/relay/blob/ee77573331057763ad97e6ba784b5654fb38c7ae/relay-metrics/src/aggregation.rs#L729).
```suggestion
    /// The MRI (metric resource identifier).
```